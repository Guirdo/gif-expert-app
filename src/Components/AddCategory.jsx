import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Form } from 'react-bootstrap';

const AddCategory = ({setCategories}) => {

    const [inputValue,setInputValue] = useState('');
    
    const handleInputChange = (e) =>{
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) =>{
        e.preventDefault();

        if(inputValue.trim().length > 2){
            setCategories(c => [inputValue,...c]);
            setInputValue('');
        }

    }

    return (
        <Form onSubmit={handleSubmit}>
            <Form.Control 
                type="text"
                value={inputValue}
                placeholder="Search anything you want"
                onChange={ handleInputChange }
            />
        </Form>
    )
}

AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired,
}

export default AddCategory;
