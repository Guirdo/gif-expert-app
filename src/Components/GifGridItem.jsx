import { Col,Image } from 'react-bootstrap';

export const GifGridItem = ( { title,url } ) => {
    return (
        <Col sm={10} md={3} lg={3} className="mx-1 mb-4 p-2 bg-white animate__animated animate__fadeIn border border-dark rounded">
            <Image 
                alt={title}
                src={url}
                fluid
            />
            <p className="mt-3 text-center">{title}</p>
        </Col>
    )
}
