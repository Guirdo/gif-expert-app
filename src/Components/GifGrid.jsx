import { Col,Row } from 'react-bootstrap';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data: images,loading } = useFetchGifs(category);

    return (
        <Col>
            <h3 className="mt-2 animate__animated animate__flipInX">{category}</h3>

            {loading && <p className="animate__animated animate__bounce">LOADING...</p>}

            <Row className="justify-content-around mt-3">
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </Row>
        </Col>
    )
}
