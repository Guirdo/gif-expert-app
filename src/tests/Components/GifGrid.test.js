import { shallow } from "enzyme";
import { GifGrid } from "../../Components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Testing GifGrid',()=>{

    const category = 'Simpson';

    test('should show the component correctly',()=>{
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('should show items when images load',()=>{
        const gifs = [{
            id: 'ABC',
            url: 'https://',
            title: 'A sample title'
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);

        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });

})