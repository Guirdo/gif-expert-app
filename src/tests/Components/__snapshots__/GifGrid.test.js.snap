// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid should show the component correctly 1`] = `
<Col>
  <h3
    className="mt-2 animate__animated animate__flipInX"
  >
    Simpson
  </h3>
  <p
    className="animate__animated animate__bounce"
  >
    LOADING...
  </p>
  <Row
    className="justify-content-around mt-3"
  />
</Col>
`;
